### PR TITLE
Use conditional compile

### PR DESCRIPTION
**Pull request by markcowl**
Failures in TypeSpecValidation are a known bug in TSV, where it validates more folders than were actually changed

### CODE REVIEW

File: eng/pipelines/templates/steps/typespec-ci.yml
Comment (mikeharder) [inline eng/pipelines/templates/steps/typespec-ci.yml:17]:
```suggestion
```

You can remove this check.  We should only call `templates/steps/typespec-ci.yml` on folders already known to contain `tspconfig.yaml`.

File: eng/pipelines/templates/steps/typespec-ci.yml
Comment (mikeharder) [inline eng/pipelines/templates/steps/typespec-ci.yml:24]:
```suggestion
```

Comment (mikeharder) [inline eng/pipelines/templates/steps/typespec-ci.yml:24]:
```suggestion
```

**Review by mikeharder — CHANGES_REQUESTED** [review]

File: eng/pipelines/templates/steps/typespec-ci.yml
Comment (markcowl) [reply eng/pipelines/templates/steps/typespec-ci.yml:20]: I think i resolved this, lmk if not let me know

File: specification/cognitiveservices/HealthInsights/healthinsights.oncophenotype/tspconfig.yaml
Comment (mikeharder) [inline specification/cognitiveservices/HealthInsights/healthinsights.oncophenotype/tspconfig.yaml:19]:
```suggestion
      description: "Azure Health Insights CancerProfiling"

```

**Review by mikeharder — APPROVED** [review]
LGTM, but please wait for review from @timotheeguerin

**Review by timotheeguerin — APPROVED** [review]

**Pull request merged**
